### PR TITLE
Search Dropdown Bug Fixes and UXA updates.

### PR DESCRIPTION
**Pull request by dimak1**
*Issues:* 
- bcgov/entity#14606
- bcgov/entity#14878

*Description of changes:*
- Search component fixes and UXA feedback updates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/search/SearchBarList.vue [hunk 1]
```diff
@@ -130,9 +133,10 @@ export default defineComponent({
         }
         return allSearchTypes
       }),
+      isPPROnly: computed((): boolean => hasPprRole.value && !(hasMhrRole.value && getFeatureFlag('mhr-ui-enabled'))),
       displayItems: [],
       displayGroup: {
-        1: false,
+        1: !(hasMhrRole.value && hasPprRole.value),
```
Comment (cameron-eyds) [inline ppr-ui/src/components/search/SearchBarList.vue]: When looking i found neither `hasMhrRole` or `hasPprRole` was set when this component mounted (the timing i was referring to previously). 
Does this not always initialize `displayGroup` as `1: true` due to that?
If it works though, great 👍

**Review by cameron-eyds — APPROVED** [review]
LGTM. Just the one comment to verify

File: ppr-ui/src/components/search/SearchBarList.vue (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/components/search/SearchBarList.vue]: I found a scenario when it doesn't work, so need to re-think this.

Comment (cameron-eyds) [reply ppr-ui/src/components/search/SearchBarList.vue]: As we discussed previously, a watcher could help with the timing.

**Review by ChasBoucher — APPROVED** [review]
Looks good.

File: ppr-ui/src/components/search/SearchBarList.vue [hunk 2]
```diff
@@ -130,9 +133,10 @@ export default defineComponent({
         }
         return allSearchTypes
       }),
+      isPPROnly: computed((): boolean => hasPprRole.value && !(hasMhrRole.value && getFeatureFlag('mhr-ui-enabled'))),
       displayItems: [],
       displayGroup: {
-        1: false,
+        1: !(hasPprRole.value && (hasMhrRole.value && getFeatureFlag('mhr-ui-enabled'))),
```
Comment (cameron-eyds) [inline ppr-ui/src/components/search/SearchBarList.vue]: How does this resolve the timing issue?

Also long term, keep in mind this flag will eventually be `true` always and we will remove this flag. Would this code remain true then? Shouldn't we be able to dictate these default values without a flag?

You seem opposed to my previous ideas of implementation, which is totally fine but i can't agree with use a 'temporary' feature flag to determine multiple values and conditions in this file. We should use the flags sparingly.

**Review by cameron-eyds — CHANGES_REQUESTED** [review]
See comment

File: ppr-ui/src/components/search/SearchBarList.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/components/search/SearchBarList.vue]: I'm not opposed, just trying to find a solution that works for all scenarios/roles/flags.

Comment (dimak1) [reply ppr-ui/src/components/search/SearchBarList.vue]: We need to use `mhr-ui-enabled` flag otherwise the dropdown behaviour won't be consistent.

Comment (cameron-eyds) [reply ppr-ui/src/components/search/SearchBarList.vue]: Please elaborate.

Comment (dimak1) [reply ppr-ui/src/components/search/SearchBarList.vue]: Messaged you offline.

**Comment by dimak1** [comment]
@cameron-eyds I refactored a bit and created a new getter prop using a feature-flag (for easier refactoring in the future). We need to check for the MHR feature-flag for the dropdown, just like we do for the other MHR-related functionality.
Also, as you mentioned onMounted() the role props are false, but I checked in onUpdated() they are set to true, right before the loading state is disabled, so we are good to use the roles props.

File: ppr-ui/src/components/search/SearchBar.vue [hunk 3]
```diff
@@ -301,9 +305,13 @@ export default defineComponent({
       categoryMessage: computed((): string => {
         return localState.validations?.category?.message || ''
       }),
+      isPPROnly: computed((): boolean => hasPprRole.value && !(hasMhrRole.value && getFeatureFlag('mhr-ui-enabled'))),
```
Comment (cameron-eyds) [inline ppr-ui/src/components/search/SearchBar.vue]: I think* the && condition is the same as the new `hasMhrRoleEnabled` property, might be useful here as well.

**Review by cameron-eyds — APPROVED** [review]
Looks good to me!
Providing the functionality is still as expected when this flag is removed 👍

File: ppr-ui/src/components/search/SearchBar.vue (hunk 3 above)
Comment (dimak1) [reply ppr-ui/src/components/search/SearchBar.vue]: Right, I meant to update it, thanks

Comment (dimak1) [reply ppr-ui/src/components/search/SearchBar.vue]: Updated

**Pull request merged**